instance/factories.py

```diff
@@ -87,8 +87,26 @@ def instance_factory(**kwargs):
     # Ensure caller provided required arguments
     assert "sub_domain" in kwargs
 
+    # HACK: This is a temporary change for BB-2558. It will be removed after upgrading MongoDB to SCRAM.
+    sandbox_settings = {
+        "FORUM_MONGO_AUTH_MECH": ":mongodb_cr",
+    }
+    configuration_extra_settings = kwargs.pop("configuration_extra_settings", "")
+    if configuration_extra_settings:
+        configuration_extra_settings = yaml.load(configuration_extra_settings, Loader=yaml.SafeLoader)
+    else:
+        configuration_extra_settings = {}
+    extra_settings = yaml.dump(
+        ansible.dict_merge(sandbox_settings, configuration_extra_settings),
+        default_flow_style=False
+    )
+    instance_kwargs = dict(
+        configuration_extra_settings=extra_settings,
+    )
+    instance_kwargs.update(kwargs)
```

Comment:
Wouldn't it make more sense to add this to instance.models.mixins.openedx_database:OpenEdXDatabaseMixin._get_mongo_settings ? Otherwise, this will add a permanent configuraiton param to the extra settings that will need to be removed once we upgrade.

Reply:
Right, that's a good point! I changed this.